### ISSUE TITLE
fields.Function doesn't pass the value

### ISSUE THREAD

**Issue by EvilTosha** [marshmallow-code/marshmallow#1176]
This is how the `Function._serialize` method looks like:

```python
    def _serialize(self, value, attr, obj, **kwargs):
        return self._call_or_raise(self.serialize_func, obj, attr)
```

As you can see, `value` is not passed inside at all. This makes it harder to work with the field (e.g. you can't write something like this: `foo = fields.Function(serialize=serialize_foo, deserialize=deserialize_foo)`). But it also makes it impossible to use it with `fields.Dict`. Consider the following code:

```python
    foo = fields.Dict(keys=str, values=fields.Function(serialize=lambda o: ..., deserialize=...)
```

`Dict._serialize` passes the arguments as follows:

```python
            for k, v in iteritems(value):
                result[keys[k]] = self.value_container._serialize(v, None, None, **kwargs)
```

So the `serialize=` lambda always receives `None` as the object.

**Comment by deckar01** [comment]
> A few methods were passing `None` into the `attr` and `obj`/`data` arguments instead of passing them through like the rest. Normalizing this did not affect the tests, but I suspect the original behavior was an obscure bug. I need to look at the implications of this change a little closer.

https://github.com/marshmallow-code/marshmallow/pull/1139#discussion_r257311386

I think #1139 would fix this.

**Comment by sloria** [comment]
Is this fixed in 3.0.0rc9?

**Comment by sloria** [comment]
Closing for now; please comment if this comes up again
